docs/compilers/CSharp/Compiler Breaking Changes - DotNet 10.md

```diff
@@ -87,3 +87,18 @@ class C
     }
 }
 ```
+
+## Warn for redundant pattern in `not ... or <redundant>` pattern
+
+***Introduced in Visual Studio 2022 version 17.13***
+
+In a disjunctive `or` pattern such as `is not null or 42` or `is not int or string`
+the second pattern is redundant and likely results from misunderstanding the precedence order
+of `not` and `or` pattern combinators.  
+The compiler provides a warning in common cases of this mistake:
+```csharp
+_ = o is not null or 42; // warning: pattern "42" is redundant
+_ = o is int or string; // warning: pattern "string" is redundant
```

Comment:
I didn't see any test for this specific scenario, do we actually warn on it?

Reply:
Sorry, that was a typo. Adding the missing `not`. We only produce the new warning in a `not ... or ...` situation.